src/verify-and-receive.ts

```diff
@@ -26,9 +27,15 @@ export async function verifyAndReceive(
     );
   }
 
-  return state.eventHandler.receive({
-    id: event.id,
-    name: event.name,
-    payload: JSON.parse(event.payload),
-  });
+  try {
+    return state.eventHandler.receive({
+      id: event.id,
+      name: event.name,
+      payload: JSON.parse(event.payload),
+    });
+  } catch (error: any) {
+    error.message = "Invalid JSON";
+    error.status = 400;
+    throw new AggregateError([error]);
+  }
```

Comment:
```suggestion
  try {
    const payload = JSON.parse(event.payload);
  } catch (error: any) {
    error.message = "Invalid JSON";
    error.status = 400;
    throw new AggregateError([error]);
  }
  
  return state.eventHandler.receive({
      id: event.id,
      name: event.name,
      payload,
  }),
```

Reply:
`payload` needs to be initialized outside of the `try/catch` to be used - you either need a `let` or to wrap it in a function that returns from within the `try/catch`

Reply:
Updated to initialize `payload` outside of the `try`/`catch`

Reply:
I oversaw your suggestion. Thank you for the suggestion ;)